pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -130,8 +131,58 @@ def cancel_invoice(self, payment_account: PaymentAccount, inv_number: str):
         # TODO not sure if direct pay can be cancelled
         """Adjust the invoice to zero."""
 
-    def get_receipt(self, payment_account: PaymentAccount, receipt_number: str, invoice_reference: InvoiceReference):
-        """Create a static receipt."""
-        # Find the invoice using the invoice_number for direct pay
+    def get_receipt(self, payment_account: PaymentAccount, pay_response_url: str, invoice_reference: InvoiceReference):
+        """Get the receipt details by calling PayBC web service."""
+        # If pay_response_url is present do all the pre-check, else check the status by using the invoice id
+        if pay_response_url is not None:
+            parsed_args = parse_url_params(pay_response_url)
+            if not parsed_args or parsed_args.get('hashValue', None) is None:
+                raise BusinessException(Error.DIRECT_PAY_INVALID_RESPONSE)
+
+            # validate if hashValue matches with rest of the values hashed
+            hash_value = parsed_args.pop('hashValue', None)[0]
+            pay_response_url_without_hash = urlencode(parsed_args)
+            if hash_value != HashingService.is_valid_checksum(pay_response_url_without_hash, hash_value):
```

Comment:
just check this statement once more..  if not  HashingService.is_valid_checksum(pay_response_url_without_hash, hash_value): might only be necesary